### PR TITLE
Handle command line arguments

### LINKED ISSUE #2: Add command line argument parsing
I've completely skipped argument parsing for now, and just read the file to compile as `argv[1]`. Let's update this to clean it up.

Options as of now:
```
-o / --output
--dump-tokens
--dump-ast
--dump-ir
--dump-register-allocation
```

### PR DESCRIPTION
**Pull request by Toafu**
Implement getopt structure

### CODE REVIEW

File: src/dcc.cpp [hunk 1]
```diff
@@ -1,18 +1,66 @@
...
 #include <string>
 
 #include "compiler.h"
 #include "macros.h"
+#include "option.h"
 #include "parser.h"
 #include "tokenizer.h"
 #include "tokens.h"
 
+std::string Option::outputFile = "out.txt";
```
Comment (danlliu) [inline src/dcc.cpp]: Can we move these to an `option.cpp` as well?

File: src/dcc.cpp [hunk 2]
```diff
@@ -1,18 +1,66 @@
...
+bool Option::dumpTokens = false;
+bool Option::dumpAST = false;
+bool Option::dumpIR = false;
+bool Option::dumpRegisterAllocation = false;
+
 int main(int argc, char** argv) {
-  if (argc != 2) ERR_AND_EXIT("Usage: " << argv[0] << " <file.dlang>");
+  if (argc < 2) ERR_AND_EXIT("Usage: " << argv[0] << "(options) <file.dlang>");
```
Comment (danlliu) [inline src/dcc.cpp]: Move this logic into a helper function that takes `argc` and `argv` and populates `Option::...`?

File: src/dcc.cpp [hunk 3]
```diff
@@ -7,13 +6,15 @@
...
 #include "parser.h"
 #include "tokenizer.h"
 #include "tokens.h"
 
 int main(int argc, char** argv) {
-  if (argc != 2) ERR_AND_EXIT("Usage: " << argv[0] << " <file.dlang>");
-  auto inFile = std::ifstream(argv[1]);
+  if (argc < 2) ERR_AND_EXIT("Usage: " << argv[0] << " (options) <file.dlang>");
+  getOptions(argc, argv);
+  auto inFile = std::ifstream(argv[argc - 1]);
```
Comment (danlliu) [inline src/dcc.cpp:17]: What about a case like `./dcc main.dlang -o main`?

Comment (Toafu) [reply src/dcc.cpp:17]: Can the input file name go anywhere or does it have to be either first or last?
Right now the best solution I can think of is iterate through all args until you find something that ends in `.dlang`

**Review by danlliu — APPROVED** [review]
LGTM

**Pull request merged**